recipes/gst-plugins-base/all/conanfile.py

```diff
@@ -0,0 +1,217 @@
+            defs["b_vscrt"] = str(self.settings.compiler.runtime).lower()
+        defs["tools"] = "disabled"
+        defs["examples"] = "disabled"
+        defs["benchmarks"] = "disabled"
+        defs["tests"] = "disabled"
+        defs["wrap_mode"] = "nofallback"
+        defs["introspection"] = "enabled" if self.options.with_introspection else "disabled"
+        defs["gl"] = "enabled" if self.options.with_gl else "disabled"
+        meson.configure(build_folder=self._build_subfolder,
+                        source_folder=self._source_subfolder,
+                        defs=defs)
+        return meson
+
+    def _copy_pkg_config(self, name):
+        root = self.deps_cpp_info[name].rootpath
+        pc_dir = os.path.join(root, 'lib', 'pkgconfig')
+        pc_files = glob.glob('%s/*.pc' % pc_dir)
+        if not pc_files:  # zlib store .pc in root
+            pc_files = glob.glob('%s/*.pc' % root)
+        for pc_name in pc_files:
+            new_pc = os.path.basename(pc_name)
+            self.output.warn('copy .pc file %s' % os.path.basename(pc_name))
+            shutil.copy(pc_name, new_pc)
+            prefix = tools.unix_path(root) if self.settings.os == 'Windows' else root
+            tools.replace_prefix_in_pc_file(new_pc, prefix)
+
+    def build(self):
+        for p in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**p)
+
+        self._copy_pkg_config("glib")
+        self._copy_pkg_config("gstreamer")
+        with tools.environment_append(VisualStudioBuildEnvironment(self).vars) if self._is_msvc else tools.no_op():
+            meson = self._configure_meson()
+            meson.build()
+
+    def _fix_library_names(self, path):
+        # regression in 1.16
+        if self.settings.compiler == "Visual Studio":
+            with tools.chdir(path):
+                for filename_old in glob.glob("*.a"):
+                    filename_new = filename_old[3:-2] + ".lib"
+                    self.output.info("rename %s into %s" % (filename_old, filename_new))
+                    shutil.move(filename_old, filename_new)
+
+    def package(self):
+        self.copy(pattern="COPYING", dst="licenses", src=self._source_subfolder)
+        with tools.environment_append(VisualStudioBuildEnvironment(self).vars) if self._is_msvc else tools.no_op():
+            meson = self._configure_meson()
+            meson.install()
+
+        self._fix_library_names(os.path.join(self.package_folder, "lib"))
+        self._fix_library_names(os.path.join(self.package_folder, "lib", "gstreamer-1.0"))
+        tools.rmdir(os.path.join(self.package_folder, "share"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "gstreamer-1.0", "pkgconfig"))
+        tools.remove_files_by_mask(self.package_folder, "*.pdb")
+
+    def package_info(self):
+        gst_plugin_path = os.path.join(self.package_folder, "lib", "gstreamer-1.0")
```

Comment:
I think this is missing setting the names for the pkg_config generator.
It looks like a lot of libraries are built.
Does this need components?

Reply:
to be honest, no idea.
I've taken bincrafters recipe which didn't have this.
for me it worked, also CI passes, so not sure if anything else is required.

Reply:
We don't test pkg_config on CI.
It will be some obscure dependency that will flag an error.
Probably best to add a `FIXME: missing pkg_config name(s) + components`.